components/roode/roode.cpp

```diff
@@ -254,164 +265,112 @@ void Roode::updateCounter(int delta) {
-       average_entry_zone_distance <= long_distance_threshold) ||
-      (average_exit_zone_distance > medium_long_distance_threshold &&
-       average_exit_zone_distance <= long_distance_threshold)) {
-    setSensorMode(3);
+  if (max > medium_distance_threshold && min <= medium_long_distance_threshold) {
+    return Ranging::MediumLong;
   }
-  if (average_entry_zone_distance > long_distance_threshold || average_exit_zone_distance > long_distance_threshold) {
-    setSensorMode(4);
+  if (max > medium_long_distance_threshold && min <= long_distance_threshold) {
+    return Ranging::Long;
   }
+  return Ranging::Longest;
 }
 
 void Roode::calibrateZones(VL53L1X_ULD distanceSensor) {
   ESP_LOGI(SETUP, "Calibrating sensor zone");
-  time_budget_in_ms = time_budget_in_ms_medium;
-  delay_between_measurements = time_budget_in_ms + 5;
-  distanceSensor.SetDistanceMode(Long);
-  sensor_status = distanceSensor.SetTimingBudgetInMs(time_budget_in_ms);
+  calibrateDistance();
+
+  entry->roi_calibration(distanceSensor, entry->threshold->idle, exit->threshold->idle, orientation_);
+  entry->calibrateThreshold(distanceSensor, number_attempts);
+  exit->roi_calibration(distanceSensor, entry->threshold->idle, exit->threshold->idle, orientation_);
+  exit->calibrateThreshold(distanceSensor, number_attempts);
 
-  if (sensor_status != VL53L1_ERROR_NONE) {
-    ESP_LOGE(CALIBRATION, "Could not set timing budget. timing_budget: %d ms, status: %d", time_budget_in_ms,
-             sensor_status);
-  }
-  int entry_threshold =
-      entry->calibrateThreshold(distanceSensor, number_attempts, max_threshold_percentage_, min_threshold_percentage_);
-  int exit_threshold =
-      exit->calibrateThreshold(distanceSensor, number_attempts, max_threshold_percentage_, min_threshold_percentage_);
-  setCorrectDistanceSettings(entry_threshold, exit_threshold);
-  if (roi_calibration_) {
-    entry->roi_calibration(distanceSensor, entry_threshold, exit_threshold, advised_sensor_orientation_);
-    entry->calibrateThreshold(distanceSensor, number_attempts, max_threshold_percentage_, min_threshold_percentage_);
-    exit->roi_calibration(distanceSensor, entry_threshold, exit_threshold, advised_sensor_orientation_);
-    exit->calibrateThreshold(distanceSensor, number_attempts, max_threshold_percentage_, min_threshold_percentage_);
-  }
-  int hundred_threshold_zone_0 = entry->getMaxThreshold() / 100;
-  int hundred_threshold_zone_1 = exit->getMaxThreshold() / 100;
-  int unit_threshold_zone_0 = entry->getMaxThreshold() - 100 * hundred_threshold_zone_0;
-  int unit_threshold_zone_1 = exit->getMaxThreshold() - 100 * hundred_threshold_zone_1;
   publishSensorConfiguration(entry, exit, true);
   App.feed_wdt();
-  if (min_threshold_percentage_ != 0) {
-    publishSensorConfiguration(entry, exit, false);
-  }
+  publishSensorConfiguration(entry, exit, false);
+}
+
+void Roode::calibrateDistance() {
+  setRangingMode(Ranging::Medium);
```

Comment:
(This is the same logic as previous). Should we not set to longest possible when trying to determine best idle distance?

Reply:
It's been updated to Longest 👍